File: UI/clinicnow_desktop/lib/models/news_item.dart
/// Mirrors the backend's `NewsItemDto`. Never carries raw image bytes - only
/// [hasImage]/[imageUrl] (the actual bytes are served from a dedicated
/// endpoint, kept off the list/detail payload).
class NewsItem {
  final int id;
  final String title;
  final String text;
  final bool hasImage;
  final String? imageUrl;
  final DateTime createdAtUtc;

  const NewsItem({
    required this.id,
    required this.title,
    required this.text,
    required this.hasImage,
    this.imageUrl,
    required this.createdAtUtc,
  });

  factory NewsItem.fromJson(Map<String, dynamic> json) {
    return NewsItem(
      id: json['id'] as int,
      title: json['title'] as String? ?? '',
      text: json['text'] as String? ?? '',
      hasImage: json['hasImage'] as bool? ?? false,
      imageUrl: json['imageUrl'] as String?,
      createdAtUtc: DateTime.parse(json['createdAtUtc'] as String),
    );
  }
}

File: ClinicNow.Services/Database/Entities/NewsItem.cs
namespace ClinicNow.Services.Database.Entities;

/// <summary>
/// A clinic-wide news/announcement item (title, text, optional image, datetime -
/// rulebook Part II §G). The image is stored as raw bytes directly in the row
/// (no separate blob storage yet - proper file-upload infrastructure with MIME/
/// magic-byte validation is Phase 6's "Medical Documentation" scope); list
/// responses never include <see cref="ImageData"/> itself (rulebook Part II §D:
/// list DTOs exclude heavy blobs) - only a <c>HasImage</c> flag, with the actual
/// bytes served from a dedicated <c>GET /api/NewsItem/{id}/image</c> endpoint.
/// </summary>
public class NewsItem
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;
    public string Text { get; set; } = string.Empty;

    public byte[]? ImageData { get; set; }
    public string? ImageContentType { get; set; }

    public DateTime CreatedAtUtc { get; set; }
}
